android/uhabits-android/src/main/java/org/isoron/uhabits/widgets/activities/NumericalCheckmarkWidgetActivity.kt

```diff
@@ -51,6 +51,14 @@ class NumericalCheckmarkWidgetActivity : Activity(), ListHabitsBehavior.NumberPi
                                   component.notificationTray)
         widgetUpdater = component.widgetUpdater
         showNumberSelector(this)
+
+        // Unlock device on clicking notification action
```

Comment:
I see that this is a repetition of the code above. Please extract them both to a common method.